tests/core/test_db.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+import pytest_asyncio
+
+
+@pytest_asyncio.fixture
+async def json_test_table(db):
+    try:
+        await db.execute("CREATE TABLE test (data JSON)")
+        yield db
+    finally:
+        await db.execute("DROP TABLE test")
+
+
+@pytest.mark.asyncio
+async def test_sql_json(db, json_test_table):
+    obj = {"a": 2, "b": 2}
+    await db.execute("INSERT INTO test VALUES(?)", (obj,))
+    row = await db.fetchone("SELECT * FROM test")
+    assert row.data == obj
+    update = {"b": 3}
+    obj |= update
+    await db.execute(f"UPDATE test SET {db.json_partial_update('data')}", (update,))
+    row = await db.fetchone("SELECT * FROM test")
```

Comment:
- can you please add a test with `WHERE` clause with JSON query?

Reply:
Ill add an additional helper in `Compat` for that aswell